service/worker/scheduler/workflow.go

```diff
@@ -253,12 +253,17 @@ func (s *scheduler) run() error {
 
 	iters := s.tweakables.IterationsBeforeContinueAsNew
 	for {
-		// TODO: use the real GetContinueAsNewSuggested
-		continueAsNewSuggested := iters <= 0 || workflow.GetInfo(s.ctx).GetCurrentHistoryLength() >= impossibleHistorySize
-		if continueAsNewSuggested && s.pendingUpdate == nil && s.pendingPatch == nil {
+		info := workflow.GetInfo(s.ctx)
+		suggestContinueAsNew := info.GetCurrentHistoryLength() >= impossibleHistorySize
+		if s.tweakables.IterationsBeforeContinueAsNew > 0 {
+			suggestContinueAsNew = suggestContinueAsNew || iters <= 0
+			iters--
+		} else {
+			suggestContinueAsNew = suggestContinueAsNew || info.GetContinueAsNewSuggested()
```

Comment:
You need to use getVersion here since this may eagerly decide to continue as new

Reply:
This workflow uses a MutableSideEffect of tweakablePolicies to get the same effect as the versioning api, but in bulk. IterationsBeforeContinueAsNew has always been > 0, we'll change it to 0 to activate this code path (in a separate PR to allow this one to go in a patch release to enable downgrade)

Reply:
Okay, I was missing that context. Thanks. LGTM.